models/model.js

```diff
@@ -8,22 +10,62 @@ exports.sendUsers = () => {
   return db.query("SELECT * FROM users").then((result) => result.rows);
 };
 
-exports.sendArticles = () => {
+exports.sendArticles = (topic, sort_by = "created_at", order = "desc") => {
+  const acceptedSort = [
+    "title",
+    "topic",
+    "author",
+    "created_at",
+    "article_img_url",
+  ];
+  const acceptedOrder = ["asc", "desc"];
+  if (!acceptedSort.includes(sort_by) || !acceptedOrder.includes(order)) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  }
+  const uniqueTopics = [];
+  testTopics.forEach((item) => {
+    if (!uniqueTopics.includes(item.slug)) {
+      uniqueTopics.push(item.slug);
+    }
+  });
```

Comment:
This method wouldn't be ideal because you're not querying the actual database - the seed data json file could become stale & out of date.

To make it truly dynamic, if the user passes a topic, do a db query to check if that topic is actually in the DB at the time of query ( rejecting the promise if it isnt)

image if at some point another topic was added to the db, this wouldn't work as it's only checking the seed data json file